test/e2e/operator.go

```diff
@@ -378,34 +385,108 @@ var _ = Describe("ARO Operator - Azure Subnet Reconciler", func() {
 })
 
 var _ = Describe("ARO Operator - MUO Deployment", func() {
-	It("must be deployed by default with FIPS crypto mandated", func(ctx context.Context) {
-		muoIsDeployed := func() (bool, error) {
-			By("getting MUO pods")
-			pods, err := clients.Kubernetes.CoreV1().Pods("openshift-managed-upgrade-operator").List(ctx, metav1.ListOptions{
-				LabelSelector: "name=managed-upgrade-operator",
-			})
-			if err != nil {
-				return false, err
-			}
-			if len(pods.Items) != 1 {
-				return false, fmt.Errorf("%d managed-upgrade-operator pods found", len(pods.Items))
-			}
+	When("we have a fresh ARO cluster", func() {
+		It("must deploy MUO by default with FIPS crypto mandated", func() {
+			fn := hasMUOFIPSCryptoMandatedFn(context.Background())
 
-			By("getting logs from MUO")
-			b, err := clients.Kubernetes.CoreV1().Pods("openshift-managed-upgrade-operator").GetLogs(pods.Items[0].Name, &corev1.PodLogOptions{}).DoRaw(ctx)
-			if err != nil {
-				return false, err
-			}
+			err := wait.PollImmediate(interval, duration, fn)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
 
-			By("verifying that MUO has FIPS crypto mandated by reading logs")
-			return strings.Contains(string(b), `msg="FIPS crypto mandated: true"`), nil
-		}
+	When("MUO is deleted on the ARO cluster", func() {
+		It("should reconcile the MUO deployment", func() {
+			ctx := context.Background()
+			fn := doesMUODeploymentExistFn(ctx)
```

Comment:
NIT: it's a little bit confusing that fn now means something different depending on were you look

Reply:
@rolandmkunkel 
I chose _fn_ because the scope of the variable is really narrow (like using _b_ for bytes) and fn just means the **f**u**n**ction that we want to "apply". In a short function like this, I think it is clear. 

But I can revert back to something like  doesMUODeploymentExistFn := doesMUODeploymentExistFn(ctx) if you want.

😃

Reply:
Agreed with Roland, I think we should rename this var.

Reply:
Done.